src/llama_stack/providers/inline/agents/meta_reference/responses/streaming.py

```diff
@@ -220,6 +239,8 @@ async def create_response(self) -> AsyncIterator[OpenAIResponseObjectStream]:
         async for stream_event in self._process_tools(output_messages):
             yield stream_event
 
+        self.ctx.chat_tool_choice = await self._process_tool_choice()
```

Comment:
why "mutate" the context? why not initialize the context correctly in the beginning?